simulator/src/eventQueue.js

```
@@ -11,93 +19,74 @@ export default class EventQueue {
     add(obj, delay) {
         if (obj.queueProperties.inQueue) {
-            obj.queueProperties.time = this.time + (delay || obj.propagationDelay);
+            const oldTime = obj.queueProperties.time;
+            const newTime = this.time + (delay || obj.propagationDelay);
+
+            // Do not reschedule if the existing event is earlier or equal
+            if (newTime >= oldTime) {
+                return;
+            }
+
+            obj.queueProperties.time = newTime;
```

ESLint

[error] Assignment to property of function parameter 'obj'.

(no-param-reassign)